web/src/js/views/read.tsx

```diff
@@ -200,54 +183,56 @@ export default class Read extends Page<Props, State> {
     }
 
     @autobind
-    private stepPage(step: number): () => void {
+    private stepPage(step: number): void {
         if (this.state.book === null) {
-            return () => null
+            return
         }
+        const book = this.state.book
 
-        return () => {
-            const book = this.state.book
+        if (!book) {
+            return
+        }
 
-            if (!book) {
-                return
-            }
+        const dst = this.state.current + step
+        if (dst < book.pages.length && dst > -1) {
+            book.current_page = dst
+            this.save()
 
-            const dst = this.state.current + step
-            if (dst < book.pages.length && dst > -1) {
-                book.current_page = dst
-                this.save()
-
-                this.setState({
-                    current: dst,
-                    book: book,
-                    width: 0,
-                    height: 0,
-                })
-            } else if (dst === book.pages.length) {
-                this.next()
-            }
+            this.setState({
+                current: dst,
+                book: book,
+                width: 0,
```

Comment:
`width` and `height` can be removed from the state, with this change they are no longer used.